packages/web-app-files/tests/unit/views/PublicFiles.spec.ts

```diff
@@ -20,4 +49,214 @@ describe('PublicFiles view', () => {
+    it('should not show the list loader', () => {
+      const wrapper = getMountedWrapper({ loading: false })
+
+      expect(wrapper.find(stubSelectors.listLoader).exists()).toBeFalsy()
+    })
+
+    describe('not found message component', () => {
+      it('should be visible if "folderNotFound" is set as "true"', () => {
+        const wrapper = getMountedWrapper({ loading: false })
+        expect(wrapper.find(stubSelectors.notFoundMessage).exists()).toBeTruthy()
+      })
+      it('should not be visible if "folderNotFound" is set as "false"', () => {
+        const wrapper = getMountedWrapper({
+          loading: false,
+          currentFolder: createFolder()
+        })
+        expect(wrapper.find(stubSelectors.notFoundMessage).exists()).toBeFalsy()
+      })
+    })
+
+    describe('no content message component', () => {
+      it('should be visible if the length of the paginated resources is zero', () => {
+        const wrapper = getMountedWrapper({
+          paginatedResources: [],
+          currentFolder: {
+            canCreate: jest.fn()
+          }
+        })
+        expect(wrapper.find(stubSelectors.noContentMessage).exists()).toBeTruthy()
+      })
+      it('should show the "call to action" information if the currentFolder has "canCreate" permission', () => {
+        stubs['no-content-message'] = false
+        const wrapper = getMountedWrapper({
+          paginatedResources: [],
+          currentFolder: createFolder({ canCreate: true })
+        })
+
+        const noContentMessage = wrapper.find('#files-public-list-empty')
+        const callToAction = noContentMessage.find('[data-testid="public-files-call-to-action"]')
+        expect(callToAction.exists()).toBeTruthy()
+        stubs['no-content-message'] = true
+      })
+      it('should not show the "call to action" information if the currentFolder doesn\'t have "canCreate" permission', () => {
+        stubs['no-content-message'] = false
+        const wrapper = getMountedWrapper({
+          paginatedResources: [],
+          currentFolder: createFolder()
+        })
+
+        const noContentMessage = wrapper.find('#files-public-list-empty')
+        const callToAction = noContentMessage.find('[data-testid="public-files-call-to-action"]')
+        expect(callToAction.exists()).toBeFalsy()
+        stubs['no-content-message'] = true
+      })
+    })
+    describe('when length of the paginated resources is greater than zero', () => {
+      const selectedResourceList = [resourceList[0], resourceList[1]]
+      const notSelectedResourceList = [resourceList[2]]
+
+      it('should load the resource table with the correct props', () => {
+        const wrapper = getMountedWrapper({
+          paginatedResources: resourceList,
+          currentFolder: createFolder()
+        })
+        const resourceTableProps = wrapper.find(stubSelectors.resourceTable).props()
+        expect(resourceTableProps).toMatchObject({
```

Comment:
Please make use snapshot testing here. Changing the object in this test whenever something changes in the code is quite cumbersome.

Reply:
fixed